### DISCUSSION TITLE
Issue with osquery agents and dashboard

### DISCUSSION THREAD

**Discussion by facyber** [Security-Onion-Solutions/securityonion#3757] [Unsupported Versions] [answered]
Hello everyone,

I updated our SO network (Manager + Heavy node) to the latest 2.3.40 and now I wanted test osquery as HIDS instead of Wazuh but I run on multiple issues.

Here are the initial steps I did on one Windows Server 2012 and one Ubuntu Server 18.04 machines:

1. Downloaded osquery from SO start page
2. Copied them to the hosts
3. Checked secret key and flag files just to be sure everything is good
4. Restarted LauncherSoLauncherSvc process on Windows (on Ubuntu machine I wasn't sure how to do it to be honest)
5. Just to be sure restarted fleet on both Manager and Heavy node.

Expected results: Fleet/osquery agents are shown in new Fleet page

Actual result: No agents are shown.

Here are additional things I did to make sure agents can communicate with the SO Manager:

1. Allowed traffic from agents to SO on port 8090. Doing telnet on port 8090 (manager) from Windows server works.
2. I did tcpdump on port 8090 and I can confirm traffic is coming from my hosts.
3. I tried to replace in flag file hostname with the IP address of the manager node like it is suggested in #2317 but that didn't helped either.

Here are some of the errors I noticed but not sure if they all are related or not:

1. Opening Osquery dashboard gives me an error `Could not locate that index-pattern-field (id: result.name.keyword)`
2. Some of the application logs on Windows machine are showed bellow [1]
3. While on Ubuntu I can see process `/usr/local/so-launcher/bin/launcher -config /etc/so-launcher/launcher.flags `running, I am not sure how can I restart it.
4. File `/opt/so/log/fleet/status.log` is empty all the time.

[1] 
caller=log.go:124 ts=2021-04-05T13:13:31.7356602Z caller=level.go:63 level=info caller=updater.go:112 msg="updater interrupted" err="enrolling host: query enrollment details, (even with retries): done trying: query enrollment details: could not query the extension manager client: i/o timeout"
caller=log.go:124 ts=2021-04-05T13:33:17.5471329Z caller=level.go:63 level=info caller=request_config.go:135 method=RequestConfig uuid=14e3e804-a713-4213-8c60-dc589ee8d291 config_size=0 reauth=false err="rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority\"" took=0s
caller=log.go:124 ts=2021-04-05T13:33:17.5490287Z caller=level.go:63 level=info caller=log.go:69 component=osquery level=stderr msg="W0405 15:33:17.549027  6224 init.cpp:589] Error reading config: error getting config: loading config failed, no cached config: transport error retrieving config: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = \"transport: authentication handshake failed: x509: certificate signed by unknown authority\"" caller=init.cpp:589

I noticed there are quite number of people who experience issue with getting osquery in WebUI, not sure if it is due the latest update or something else. Hopefully somebody have solution, as we would really like to replace Wazuh with Osquery (or maybe to use them both), as we are interested in sort of an asset management that fleet provides.

Thanks,
facyber

**Comment by facyber** [comment]
Just to mention, not sure if it is important or not, but during initial setup, I chose specific DNS for accessing web UI.

**Comment by defensivedepth** [comment, marked as answer]
Do you have an SSL proxy in between the endpoints and the Manager or between the endpoints and the Internet? Something that would be breaking SSL connection?

**Reply by facyber** [reply]
Hmm well since setting up Let's Encrypt was not applicable by our standards, it is done in a bit different way (I was following [this thread](https://github.com/Security-Onion-Solutions/securityonion/discussions/2883#discussioncomment-343221), we then configured other host for generating TLS certificates and then copying the files to the SO manager note in _/etc/pki/_, managerssl.crt and managerssl.key files.

But there shouldn't be any other proxy, neither I have anything in /etc/environment.

**Reply by defensivedepth** [reply]
yep that would be the issue. Also see that thread where a poster says:

`Note: I found an issue with osquery when the cert was replaced, I'm still trying to track this but I think ca.crt/ca.key have to match the LE one as well.`

I have created an issue to track documentation of this configuration and how to get osquery agents to work with this config as welll:  https://github.com/Security-Onion-Solutions/securityonion/issues/3771

**Reply by facyber** [reply]
Hah it seems I missed that notification...

Thank you very much will track it then.

**Reply by facyber** [reply]
Just a small update:  I replaced certificate with the LE and there are no more error logs that says there is issue with certificate, and I also tested with certs generated by SO, the issue is still present. I tried restarting both nginx and fleet on manager node.

caller=log.go:124 ts=2021-04-08T17:31:38.6914365Z caller=level.go:63 level=info caller=updater.go:112 msg="updater interrupted" err="enrolling host: query enrollment details, (even with retries): done trying: query enrollment details: could not query the extension manager client: i/o timeout"
caller=log.go:124 ts=2021-04-08T17:31:35.9806096Z caller=level.go:63 level=info caller=runtime.go:649 path="C:\\Program Files\\Kolide\\Launcher-so-launcher\\bin\\osqueryd.exe-updates\\1617903036\\osqueryd.exe" sizeBytes=21644960 mode=-rw-rw-rw- sha256=9b2f3b75fb13f9447dbfd719d759650d2e82fc94d8ad4e9505e1114617f551ca msg="Error running osquery command" err="exit status 1"

These are logs from agent, while there is nothing still in /opt/so/log/fleet/status.log file.

Edit: after I replaced cert on Linux host with LE's, I see Linux host in Fleet, but Windows host still have issue. So half way there. :)

**Reply by facyber** [reply]
Ok last post here and will stop spamming.

It seems it works on other Windows host with newer OS, so it could be OS related. 

Still need to run few more tests, but will try to automate the process and help with the documentation if I can.

**Reply by baylorex97** [reply]
Did you find a resolution to this issue?  I am getting the same error, but just on one Windows host.  I have attached the debug log.  Thanks.

could not query the extension manager client: i/o timeout

[OSQueryDebug.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7590197/OSQueryDebug.txt)

**Reply by facyber** [reply]
> Did you find a resolution to this issue?  I am getting the same error, but just on one Windows host.  I have attached the debug log.  Thanks.
> 
> could not query the extension manager client: i/o timeout
> 
> [OSQueryDebug.txt](https://github.com/Security-Onion-Solutions/securityonion/files/7590197/OSQueryDebug.txt)
> 

As far as I remember, no. We moved to Wazuh and that was it. I am no longrr working for the company where I worked on SO.

**Comment by baylorex97** [comment]
Thanks for the response, guess I will move in the same direction.
